app/src/test/java/com/kickstarter/viewmodels/ProjectViewModelTest.kt

```diff
@@ -466,11 +504,109 @@ class ProjectViewModelTest : KSRobolectricTestCase() {
         this.projectTest.assertValueCount(2)
 
         this.vm.inputs.pledgeSuccessfullyCancelled()
-        this.showRewardsFragment.assertValue(false)
+        this.expandPledgeSheet.assertValue(false)
         this.showCancelPledgeSuccess.assertValueCount(1)
         this.projectTest.assertValueCount(3)
     }
 
+    @Test
+    fun testManagePledgeMenuIsVisible_whenProjectBacked() {
+        setUpEnvironment(environmentWithNativeCheckoutEnabled())
+
+        // Start the view model with a backed project
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.backedProject()))
+
+        this.managePledgeMenuIsVisible.assertValue(true)
+    }
+
+    @Test
+    fun testManagePledgeMenuIsVisible_whenProjectNotBacked() {
```

Comment:
I've been _trying_ to structure tests like test[output]_when[condition]. Does that make sense? Nino and I have been flopping around about it.

Reply:
I think both make a lot of sense. Personally when I read tests that I'm unfamiliar with, I appreciate being able to get as much context from it without having to read through the whole test body. It saves me some time and allows me to jump back to the code.

Reply:
Yeah, we've been discussing ways to make the tests more readable for people that are unfamiliar with the codebase. The pattern @eoji mentioned above is used on iOS too. Once you get into inputs/outputs it will make more sense, but maybe a discussion about it would be worth it.